llvm/test/CodeGen/RISCV/cheri/compress.ll

```diff
@@ -0,0 +1,118 @@
+; This test is designed to run twice, once with function attributes and once
+; with target attributes added on the command line.
+; See compress.ll in the folder above
+;
+; RUN: cat %s > %t.tgtattr
+; RUN: echo 'attributes #0 = { nounwind }' >> %t.tgtattr
+; RUN: %riscv32_cheri_purecap_llc -mattr=+c,+xcheri,+cap-mode -filetype=obj < %t.tgtattr \
+; RUN:   | llvm-objdump -d -M no-aliases --syms - | FileCheck %s
+; RUN: %riscv64_cheri_purecap_llc -mattr=+c,+xcheri,+cap-mode -filetype=obj < %t.tgtattr \
+; RUN:   | llvm-objdump -d -M no-aliases --syms - | FileCheck %s
+; RUN: %riscv64_cheri_purecap_llc -mattr=+c,+xcheri,+cap-mode,+xcheri-norvc -filetype=obj < %t.tgtattr \
+; RUN:   | llvm-objdump -d -M no-aliases --syms - | FileCheck %s --check-prefix=NORVC
+
+; RUN: cat %s > %t.fnattr
+; RUN: echo 'attributes #0 = { nounwind "target-features"="+c,+xcheri,+cap-mode" }' >> %t.fnattr
+; RUN: %riscv32_cheri_purecap_llc -filetype=obj < %t.fnattr \
+; RUN:   | llvm-objdump -d --mattr=+c -M no-aliases --syms - | FileCheck %s
+; RUN: %riscv64_cheri_purecap_llc -filetype=obj < %t.fnattr \
+; RUN:   | llvm-objdump -d --mattr=+c -M no-aliases --syms - | FileCheck %s
+; RUN: cat %s > %t.fnattr
+; RUN: echo 'attributes #0 = { nounwind "target-features"="+c,+xcheri,+cap-mode,+xcheri-norvc" }' >> %t.fnattr
+; RUN: %riscv64_cheri_purecap_llc -filetype=obj < %t.fnattr \
+; RUN:   | llvm-objdump -d --mattr=+c -M no-aliases --syms - | FileCheck %s  --check-prefix=NORVC
+
+; Basic check that we can use CHERI compressed instructions
+
+; Check the size of the functions to check whether we used compressed instructions:
+; CHECK-LABEL: SYMBOL TABLE:
+; CHECK: 0000001a loadstore
+; CHECK: 00000028 call
+; NORVC-LABEL: SYMBOL TABLE:
+; NORVC: 00000030 loadstore
+; NORVC: 00000034 call
+
+define i32 @loadstore(i32 addrspace(200)* %heapint, i8 addrspace(200)* addrspace(200)* %heapptr) addrspace(200) #0 {
```

Comment:
Why heap? Could be anything... Also heapint is a bit confusing given it is still a pointer, just a pointer to an int supposedly on the heap.

Reply:
Yeah that's a bad name, I started off with just an alloca, so went for heap there. will fix.